src/types/line.js

```diff
@@ -1,43 +1,58 @@
 import {Element} from 'chart.js';
-import {PI, toRadians, toDegrees, toPadding} from 'chart.js/helpers';
-import {EPSILON, clamp, scaleValue, measureLabelSize, getRelativePosition, setBorderStyle, setShadowStyle, getElementCenterPoint, retrieveScaleID, getDimensionByScale} from '../helpers';
+import {PI, toRadians, toDegrees, toPadding, distanceBetweenPoints} from 'chart.js/helpers';
+import {EPSILON, clamp, rotated, measureLabelSize, getRelativePosition, setBorderStyle, setShadowStyle, getElementCenterPoint, toPointOption, getSize, resolveLineProperties} from '../helpers';
 
 const pointInLine = (p1, p2, t) => ({x: p1.x + t * (p2.x - p1.x), y: p1.y + t * (p2.y - p1.y)});
 const interpolateX = (y, p1, p2) => pointInLine(p1, p2, Math.abs((y - p1.y) / (p2.y - p1.y))).x;
 const interpolateY = (x, p1, p2) => pointInLine(p1, p2, Math.abs((x - p1.x) / (p2.x - p1.x))).y;
 const sqr = v => v * v;
 const rangeLimit = (mouseX, mouseY, {x, y, x2, y2}, axis) => axis === 'y' ? {start: Math.min(y, y2), end: Math.max(y, y2), value: mouseY} : {start: Math.min(x, x2), end: Math.max(x, x2), value: mouseX};
+// http://www.independent-software.com/determining-coordinates-on-a-html-canvas-bezier-curve.html
+const coordInCurve = (start, cp, end, t) => (1 - t) * (1 - t) * start + 2 * (1 - t) * t * cp + t * t * end;
+const pointInCurve = (start, cp, end, t) => ({x: coordInCurve(start.x, cp.x, end.x, t), y: coordInCurve(start.y, cp.y, end.y, t)});
+const coordAngleInCurve = (start, cp, end, t) => 2 * (1 - t) * (cp - start) + 2 * t * (end - cp);
+const angleInCurve = (start, cp, end, t) => -Math.atan2(coordAngleInCurve(start.x, cp.x, end.x, t), coordAngleInCurve(start.y, cp.y, end.y, t)) + 0.5 * PI;
 
 export default class LineAnnotation extends Element {
 
   inRange(mouseX, mouseY, axis, useFinalPosition) {
     const hBorderWidth = this.options.borderWidth / 2;
     if (axis !== 'x' && axis !== 'y') {
-      const epsilon = sqr(hBorderWidth);
       const point = {mouseX, mouseY};
+      const {path, ctx} = this;
+      if (path) {
+        setBorderStyle(ctx, this.options);
+        const result = ctx.isPointInStroke(path, mouseX, mouseY) || isOnLabel(this, point, useFinalPosition);
```

Comment:
I'm not 100% the mouseX/mouseY are canvas coordinates (when dpr !== 1), but they should be.

Reply:
Apologize @kurkle but I don't understand. What  does `(when dpr !== 1),` mean?

Reply:
DevicePixelRatio, eg. Retina display

Reply:
Ok, forgive my lack of knowledge... I can do some tests with different dpr and to see if it works. I'll check also the documentation of the `Path` to see if the topic is mentioned. But do you have any experience with `Path`s where dpr doesn't work?

Reply:
@kurkle I don't know if I did a thorough test, changing Karma config, and enabling DPR to 2

```javascript
    customLaunchers: {
      chrome: {
        base: 'Chrome',
        flags: [
          '--disable-accelerated-2d-canvas',
          '--disable-background-timer-throttling',
          '--disable-backgrounding-occluded-windows',
          '--disable-renderer-backgrounding'
        ]
      },
      firefox: {
        base: 'Firefox',
        prefs: {
          'layers.acceleration.disabled': true,
          'layout.css.devPixelsPerPx': 2.0 // <--- change pixel ratio
        }
      }
    },
```

and running the test on FF (`npm run dev:ff`), no issue at all.

I don't see any other way to test it :(.

What do u think?

Reply:
Not sure if that works, but I think quickest test is to run a sample page in browser, zoom in and see if it works properly

Reply:
thanks! I tested and you're right. It doesn't work. uhm.... without Path2D it's very difficult to implement a curve... Let me investigate or do you know any trick about that?

Reply:
Found it!

Reply:
![image](https://user-images.githubusercontent.com/11741250/193045086-0c85f898-4b4e-4da8-956d-6e532831cd5c.png)

Reply:
Should not be that simple, was you chart at the top left corner of the window? :)

Reply:
https://github.com/chartjs/Chart.js/blob/52cf8e8a94130d63b7618751fe0df871ca1af222/src/helpers/helpers.dom.js#L94-L123 that should do it

Reply:
WOW!!! Great! I wasn't aware the getRelativePosition is also managing the DPR. Every time I'm learning some thing new with you!!!!! UR awesome! I really appreciate. I ca tell you only thank youu! 
But coming back to the topic, I need to find a way to get the chart, because is needed to getRelativePosition and I don't have it in inRange method... let me try... I'll find a way (that maybe you don't like it :) )

Reply:
I have it, in the element context!!!

Reply:
Uhm... it doesn't work. Inside the inRange I'm calling getRelativePosition but, tracking also in Chart.js, the original event was already "translated" by  getRelativePosition . Therefore applying it again, I'm quite sure the new {x, y} is not correct.

Reply:
Code:

```javascript
        const {chart} = this.$context;
        const {x, y} = getRelativePosition({clientX: mouseX, clientY: mouseY}, chart);
        const result = ctx.isPointInStroke(path, x, y) || isOnLabel(this, point, useFinalPosition);
```

Instead, it's continuing working multiplying the mouseX and mouseY to DPR..

Reply:
The original event: `clientX: 369, clientY: 71`
In range argument: `mouseX: 361,mouseY: 63` (after getRelativePosition invoked in CHART.js)
after my call to getRelativePosition: `x: 353,y: 55`

mouseX, mouseY multiplied by DPR: `mx: 481.3333333333333, my: 84, dpr: 1.3333333333333333` (it works)

Reply:
Maybe I'm wrong but getRelativePosition is dividing by DPR (and this is what we are receiving as mouseX and mouseY). Therefore to come back the original position, we should multiply by dpr.... but, repeating myself, maybe I'm wrong.

Reply:
hmm, ok. Maybe your original solution is correct then. There must be something different in `Path`

Reply:
yes, I think so :(